wcfsetup/install/files/lib/acp/form/LanguageAddForm.class.php

```diff
@@ -103,149 +73,105 @@ public function readParameters()
-        }
-        if (isset($_POST['sourceLanguageID'])) {
-            $this->sourceLanguageID = \intval($_POST['sourceLanguageID']);
-        }
-        if (isset($_POST['locale'])) {
-            $this->locale = $_POST['locale'];
-        }
-    }
-
-    /**
-     * @inheritDoc
-     */
-    public function validate()
+    protected function createForm()
     {
-        parent::validate();
-
-        // language name
-        if (empty($this->languageName)) {
-            throw new UserInputException('languageName');
-        }
-
-        // country code
-        if (empty($this->countryCode)) {
-            throw new UserInputException('countryCode');
-        }
-
-        // language code
-        $this->validateLanguageCode();
+        parent::createForm();
 
-        // source language id
-        $this->validateSource();
-
-        if (!isset($this->locales[$this->locale])) {
-            throw new UserInputException('locale');
-        }
-    }
+        $this->languages = LanguageFactory::getInstance()->getLanguages();
 
-    /**
-     * Validates the language code.
-     */
-    protected function validateLanguageCode()
-    {
-        if (empty($this->languageCode)) {
-            throw new UserInputException('languageCode');
-        }
-        if (LanguageFactory::getInstance()->getLanguageByCode($this->languageCode)) {
-            throw new UserInputException('languageCode', 'notUnique');
-        }
+        $locales = [
+            '0' => WCF::getLanguage()->get('wcf.global.noSelection'),
+        ];
+        $locales = \array_merge($locales, $this->locales);
+
+        $this->form->appendChildren([
+            FormContainer::create('data')
+                ->appendChildren([
+                    TextFormField::create('languageName')
+                        ->label('wcf.global.name')
+                        ->description('wcf.acp.language.name.description')
+                        ->maximumLength(255)
+                        ->required(),
+                    TextFormField::create('languageCode')
+                        ->label('wcf.acp.language.code')
+                        ->description('wcf.acp.language.code.description')
+                        ->maximumLength(20)
+                        ->required()
+                        ->addValidator(new FormFieldValidator('unique', function (TextFormField $formField) {
+                            if (!empty($formField->getValidationErrors())) {
+                                return;
+                            }
+                            if (
+                                $this->formObject instanceof Language
+                                && \mb_strtolower($this->formObject->languageCode) === $formField->getValue()
+                            ) {
+                                return;
+                            }
+
+                            if (LanguageFactory::getInstance()->getLanguageByCode($this->languageCode)) {
```

Comment:
`languageCode` is an undefined property.
```suggestion
                            if (LanguageFactory::getInstance()->getLanguageByCode($formField->getValue())) {
```
?